### PR TITLE
Added support for progress.

### PR DESCRIPTION
**Pull request by allan-walker**
This pr adds three new props to <Scrollama />:
progressmode: bool,
onStepProgress: callback function, 
threshold: int

onStepProgress has three parameters:
element (step),
data,
progress (float between 0-1)

### CODE REVIEW

File: src/lib/Scrollama.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ const getPageHeight = () => {
 class Scrollama extends PureComponent {
   constructor(props) {
     super(props);
-    const { offset, debug, children, onStepEnter, onStepExit } = this.props;
+    const { offset, debug, children, onStepEnter, onStepExit, progressmode, onStepProgress, threshold } = this.props;
```
Comment (jsonkao) [inline src/lib/Scrollama.js]: `progressmode` should be in camelCase. If you don't want there to be a naming conflict, I suggest you name it something else.

File: src/lib/Scrollama.js [hunk 2]
```diff
@@ -66,6 +70,12 @@ class Scrollama extends PureComponent {
     this.handleEnable(true);
   }
 
+  componentWillMount() {
+    this.setState({
+      progressThreshold: Math.max(1, +this.state.threshold),
```
Comment (jsonkao) [inline src/lib/Scrollama.js]: This can be done in the constructor, no?

**Comment by schlesingermatthias** [comment]
@allan-walker @jsonkao any update on this PR?

**Comment by allan-walker** [comment]
@schlesingermatthias @jsonkao I believe the requested changes have been made and ready for @jsonkao to review and merge.

**Comment by jefffriesen** [comment]
This would be great to get this merged

**Pull request closed without merging**